src/duckling/controls/json.component.ts

```diff
@@ -102,47 +129,35 @@ export class JsonComponent {
         }
         this.valueChanged.emit(immutableAssign(this.value, patch));
     }
-
-    get jsonKeys() : string[] {
-        let keys : string[] = [];
-        for (let key in this.value) {
-            keys.push(key);
-        }
-        return keys;
+    
+    onArrayAddClicked(key : string) {
+        let patchArray = [];
+        patchArray[this.value[key].length] = getDefaultForSchema((this.schema[key] as JsonSchema[])[0] as JsonSchema);
```

Comment:
D:\Jeff\Code\ILikeDucks\duckling\node_modules\@angular\core\bundles\core.umd.js:2843 TypeError: Cannot read property 'keys' of undefined
    at JsonSchemaEditComponent.getKeyIndices (D:\Jeff\Code\ILikeDucks\duckling\build\duckling\controls\json-schema-edit.component.js:164)
    at DebugAppView.View_JsonSchemaEditComponent0.detectChangesInternal (component.ngfactory.js:84)
    at DebugAppView.AppView.detectChanges (D:\Jeff\Code\ILikeDucks\duckling\node_modules\@angular\core\bundles\core.umd.js:9354)
    at DebugAppView.detectChanges (D:\Jeff\Code\ILikeDucks\duckling\node_modules\@angular\core\bundles\core.umd.js:9447)
    at DebugAppView.View_CustomAttributesComponent0.detectChangesInternal (component.ngfactory.js:274)
    at DebugAppView.AppView.detectChanges (D:\Jeff\Code\ILikeDucks\duckling\node_modules\@angular\core\bundles\core.umd.js:9354)
    at DebugAppView.detectChanges (D:\Jeff\Code\ILikeDucks\duckling\node_modules\@angular\core\bundles\core.umd.js:9447)
    at DebugAppView.View_CustomAttributesComponent_Host0.detectChangesInternal (host.ngfactory.js:29)
    at DebugAppView.AppView.detectChanges (D:\Jeff\Code\ILikeDucks\duckling\node_modules\@angular\core\bundles\core.umd.js:9354)
    at DebugAppView.detectChanges (D:\Jeff\Code\ILikeDucks\duckling\node_modules\@angular\core\bundles\core.umd.js:9447)